overlord/snapstate/snapstate.go

```diff
@@ -314,10 +311,73 @@ func Remove(s *state.State, name string) (*state.TaskSet, error) {
 	return full, nil
 }
 
-// Rollback returns a set of tasks for rolling back a snap.
+// Revert returns a set of tasks for reverting to the pervious version of the snap.
 // Note that the state must be locked by the caller.
-func Rollback(s *state.State, snap, ver string) (*state.TaskSet, error) {
-	return nil, fmt.Errorf("rollback not implemented")
+func Revert(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+
+	pi := snapst.PreviousSideInfo()
+	if pi == nil {
+		return nil, fmt.Errorf("no revision to revert to")
+	}
+	return revertToRevision(s, name, pi.Revision)
+}
+
+func revertToRevision(s *state.State, name string, rev snap.Revision) (*state.TaskSet, error) {
```

Comment:
It is similar in some sense but also slightly different. A revert is an install without `mount-snap` and without `data-copy` tasks. So instead of (prepare,mount,copy-data,setup-profiles,link) it is (prepare, setup-profiles, link). I made revertToRevision use doInstall now in 46634f0.